### PR TITLE
More sounds and polish

### PR DESCRIPTION
**Pull request by FredTheDino**
Sounds for a lot of things! Not sure about the background sound though - it's trubblesome to have music since we cannot pause it and the server will hear it aswell. Might require fixing.

But I do like that we have MORE PARTICLES!!! :D

### CODE REVIEW

**Review by th3tard1sparadox — CHANGES_REQUESTED** [review]
I think it's an improvement, but I would like the cheering sounds for hitting a goal to be slightly longer.

On the background noise: it's nice but feels a bit weird when you're in the menu, only needs to be fixed if we have time for it though.

File: entity.sy [hunk 1]
```diff
@@ -149,26 +149,38 @@ player_new :: fn pos: (float, float), team: int, owner: int -> Player {
...
+
+            if dot(input.dir, util.dir(self.rot)) < 0.8 && input_mag > 0.0 {
+                // TODO(ed): Spray particles here
+                assets.random_skate()
+                particles.turn' self, rotation
+            }
+
             self.rot += rotation * delta
             self.rot = rem' self.rot, 2. * pi
+
```
Comment (EMattfolk) [inline entity.sy]: :open_mouth: 
```suggestion
```

File: entity.sy [hunk 2]
```diff
@@ -149,26 +149,38 @@ player_new :: fn pos: (float, float), team: int, owner: int -> Player {
...
             }
 
             force :: 3.0
+            input_mag :: magnitude(input.dir)
             if !ignore_movement {
-                self.acc += force * magnitude' input.dir
+                self.acc += force * input_mag
             }
 
+
```
Comment (EMattfolk) [inline entity.sy]: :open_mouth: 
```suggestion
```

**Review by EMattfolk — APPROVED** [review]
he

**Review by th3tard1sparadox — APPROVED** [review]

**Pull request merged**